src/main/java/com/softserve/actent/model/entity/Event.java

```diff
@@ -0,0 +1,124 @@
+package com.softserve.actent.model.entity;
+
+import com.fasterxml.jackson.annotation.JsonManagedReference;
+import com.softserve.actent.constant.StringConstants;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.validator.constraints.Length;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.ForeignKey;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotBlank;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Set;
+
+@Data
+@NoArgsConstructor
+@Entity
+@Table(name = "events")
+public class Event {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NonNull
+    @NotBlank(message = StringConstants.TITLE_SHOULD_NOT_BE_BLANK)
+    @Length(max = 100, message = "Too long")
+    @Column(nullable = false, length = 100)
+    private String title;
+
+    @Length(max = 500, message = "Too long")
+    @Column(length = 500)
+    private String description;
+
+    @CreationTimestamp
+    @Column(name = "creation_date")
+    private Timestamp creationDate;
+
+    @NonNull
+    @Column(name = "start_date", nullable = false)
+    private Long startDate;
+
+    @NonNull
+    @Column(nullable = false)
+    private Long duration;
+
+    @NonNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(nullable = false, foreignKey = @ForeignKey(name = "FK_event_to_creator_id"))
+    private User creator;
+
+    @NonNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "location_id", nullable = false, foreignKey = @ForeignKey(name = "FK_event_to_location_id"))
+    private Location address;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "image_id", foreignKey = @ForeignKey(name = "FK_event_to_image_id"))
+    private Image image;
+
+    private Integer capacity;
+
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "assignedEvent")
+    private List<Equipment> equipments;
```

Comment:
Why not Set? Also we need to initialize collections.

Reply:
https://docs.jboss.org/hibernate/orm/3.6/reference/en-US/html/collections.html you may take a look into documentation for references. There are some info about initialization with "new" and lot of examples.

Reply:
Ask Igor!

Reply:
?
I've provided examples from documentation, you're implying that there are wrong approaches?

Reply:
Ask Igor!

Reply:
I can understand you from the first time, but author of this code is you, not Ihor. If you do not know why you are choosing one approach over another, ask Ihor for yourself and do not send me to him, thank you.

Reply:
Why are you resolving conversations without any changes or commentary?

Reply:
Because I do not accept your opinion!

Reply:
So, you may provide your, but not just sneeze on it. Conversation will be resolved only after consent.

Reply:
Yes) Ask Igor and that will end)

Reply:
Is Ihor stated anywhere here as author of this entity? Why should I bother him for your code?